javalib/src/main/scala/java/util/Formatter.scala

```diff
@@ -28,4 +34,18 @@ class Formatter {
   override def toString: String                              = ???
 }
 
-object Formatter
+object Formatter {
+
+  private class RegExpExtractor(val regexp: Regex) {
+    def unapply(str: String): Option[Regex.Match] = {
+      regexp.findFirstMatchIn(str)
+    }
+  }
+
+  private val RegularChunk = new RegExpExtractor("""^[^\x25]+""".r)
```

Comment:
Aw shot I did not notice that the [Scala.js](https://github.com/scala-js/scala-js/blob/master/javalib/src/main/scala/java/util/Formatter.scala#L273) implementations use Regex. 

I think we should implement it without regex for now. It's also better because it will reduce the coupling in java.util.*